README.md

```diff
@@ -1,26 +1,56 @@
 
-# About
+# lsif-util
 
-Scripts to help [LSIF](https://github.com/Microsoft/language-server-protocol/blob/master/indexFormat/specification.md) developers. Featuring:
+Scripts to help [LSIF](https://github.com/Microsoft/language-server-protocol/blob/master/indexFormat/specification.md) developers.
 
-* Validation
-* Graph visualization
-* Quick search
+## What's new
 
-# Getting Started
+* _Up to date!_ Support for [JSON Lines](http://jsonlines.org/) with the `--inputFormat` tag
+* _Cleaning up!_ Standard filtering for all tools (no more unnecessary searching)
 
-* `> git clone this repository`
-* `> cd lsif-util`
-* `> npm install`
-* `> npm run compile`
+## What's next
 
-# Validation
+* _We are moving!_ The lsif-util tools will soon migrate to [lsif-node](https://github.com/microsoft/lsif-node)
+* _Getting official!_ Global npm package is on the way
 
-`> node .\lib\validate.js [options]`
+## Usage
 
-| Option            | Default     | Description                                             |
-|-------------------|-------------|---------------------------------------------------------|
-| --inputPath or -p | ./lsif.json | Path to input file (JSON)                               |
+``` bash
+git clone https://github.com/jumattos/lsif-util.git
+cd lsif-util
+npm install
+npm run compile
+node .\lib\main.js [validate|visualize] [file] --inputFormat [line|json] [--stdin] [filters]
+```
+
+| Option        | Description                                    | Default |
+|---------------|------------------------------------------------|---------|
+| --inputFormat | Specify input format (choices: "line", "json") | line    |
+| --stdin       | Read from standard input                       | false   |
+| --id          | Filter by id                                   | []      |
+| --inV         | Filter by inV                                  | []      |
+| --outV        | Filter by outV                                 | []      |
+| --type        | Filter by type                                 | []      |
+| --label       | Filter by label                                | []      |
+| --property    | Filter by property                             | []      |
+| --regex       | Filter by regex                                | none    |
+
+Validating outgoing edges from vertex 1:
+``` bash
+node .\lib\main.js validate .\example\line.json --outV 1
+```
+
+Visualizing ranges:
```

Comment:
It can be slightly confusing for a new user to see "visualizing ranges" here, when the "visualization" section is posted after. Would it be useful to introduce what visualization means before sharing usage instructions for it?

Reply:
Good catch!

Reply:
There is a circular dependency between "usage" and "validation/visualization". Which one should be first?

Reply:
I tried a new approach with Usage > Validation > Visualization > Filters. Let me know if that sounds better.